pkg/api/v1/types.go

```diff
@@ -1981,34 +1981,89 @@ type SecretList struct {
 	Items []Secret `json:"items" description:"items is a list of secret objects; see http://releases.k8s.io/HEAD/docs/user-guide/secrets.md"`
 }
 
-// Type and constants for component health validation.
+type ComponentType string
+
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler"
+)
+
+type ComponentSpec struct {
+	Type    ComponentType `json:"type" description:"type of the component"`
+	Address string        `json:"address,omitempty" description:"address to reach the component (scheme, host, port, path); address paths determined by component type"`
+}
+
+type ComponentPhase string
+
+const (
+	ComponentPending    ComponentPhase = "Pending"
+	ComponentRunning    ComponentPhase = "Running"
+	ComponentTerminated ComponentPhase = "Terminated"
+)
+
 type ComponentConditionType string
 
-// These are the valid conditions for the component.
 const (
-	ComponentHealthy ComponentConditionType = "Healthy"
+	ComponentRunningHealthy    ComponentConditionType = "Healthy"
```

Comment:
"Heathy=True" means the same thing a `/healthz` ok means: 

1. Instance is Up (without error)
1. Dependencies are Up (at least enough for this component to function)
1. Ready to respond to requests

How is that ambiguous?

In the case of heartbeating, the networking implications are slightly different than if it were from a healthcheck:
- Healthy from healthcheck would mean the apisever can reach the component instance
- Healthy from heartbeat would mean the component instance can reach the apiserver